### PR TITLE
TANF Formula for is_tanf_eligible

### LINKED ISSUE #336: Add TANF IL eligibility based on ages and pregnancy

### PR DESCRIPTION
**Pull request by yvvijay121**
# New formula for TANF eligibility

- [x] Label field added
- [x] Documentation field added
- [x] Unit field added
- [x] Default value field added if relevant
- [x] Variable name follows conventions
- [x] Unit test(s) added

## What's changed

- Added formula to `is_tanf_eligible` variable in `tanf.py`
- reorganizing TANF tests to be aligned with the formulas for each TANF variable in `tanf.py`
- Fixes #336

### CODE REVIEW

**Review by MaxGhenis — APPROVED** [review]
Looks good! Let's just add a couple tests for a SPM unit with multiple people, e.g. two ineligible, 1 eligible one ineligible, two eligible.

File: openfisca_us/tests/policy/baseline/hhs/tanf/tanf_amount_if_eligible.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  absolute_error_margin: 0.01
+  input:
+    tanf_max_amount: 1000
+    tanf_countable_income: 750
+  output:
+    tanf_amount_if_eligible: 1000 - 750
+
+- name: "TANF: Has given maximum benefit amount and has larger countable income"
+  period: 2022
+  absolute_error_margin: 0.01
```
Comment (nikhilwoodruff) [inline openfisca_us/tests/policy/baseline/hhs/tanf/tanf_amount_if_eligible.yaml:12]: Does OpenFisca have some small rounding issue here? Seems inconsequential but just curious

**Review by nikhilwoodruff — APPROVED** [review]
Looks good to me, thanks @yvvijay121

**Comment by nikhilwoodruff** [comment]
@yvvijay121 I think the tests are failing because some of the tests are missing a `period: 2022` assignment, could you add those?

File: openfisca_us/tests/policy/baseline/hhs/tanf/tanf_amount_if_eligible.yaml (hunk 1 above)
Comment (MaxGhenis) [reply openfisca_us/tests/policy/baseline/hhs/tanf/tanf_amount_if_eligible.yaml:12]: It did for some other test so I had to relax it in https://github.com/PolicyEngine/openfisca-us/commit/b99f2c46ef5ecce768658d6f8e0a8ddef20574ed

**Pull request merged**